vstruct/primitives.py

```diff
@@ -123,30 +123,47 @@ def __hash__(self):
 
 
 num_fmts = {
+    # big endian unsigned
     (True, 1): '>B',
     (True, 2): '>H',
     (True, 4): '>I',
     (True, 8): '>Q',
+
+    # little endian unsigned
     (False, 1): '<B',
     (False, 2): '<H',
     (False, 4): '<I',
     (False, 8): '<Q',
 }
 
+signed_fmts = {
+    # big endian signed
+    (True, 1): '>b',
+    (True, 2): '>h',
+    (True, 4): '>i',
+    (True, 8): '>q',
+
+    # little endian signed
+    (False, 1): '<b',
+    (False, 2): '<h',
+    (False, 4): '<i',
+    (False, 8): '<q',
+}
+
 
 class v_number(v_prim):
     _vs_length = 1
 
     def __init__(self, value=0, bigend=False, enum=None):
         v_prim.__init__(self)
         self._vs_bigend = bigend
-        self._vs_value = value
         self._vs_enum = enum
         self._vs_length = self.__class__._vs_length
         self._vs_fmt = num_fmts.get((bigend, self._vs_length))
 
         # TODO: could use envi.bits, but do we really want to dep on envi?
         self.maxval = (2 ** (8 * self._vs_length)) - 1
+        self.vsSetValue(value)
```

Comment:
is this a preference?  or was it causing issues?

Reply:
It's for consistency more than anything. If you throw a value into the constructor, we just set it and forget it. But if you use vsSetValue, it'll get masked down with `self.maskval`, and for `v_snumber`, it'll do a slightly more complex masking. I figured it'd be good to be consistent here whether you use the constructor or vsSetValue.

Reply:
i understand that value.  i'd mention speed, but i'm hopeful that this method isn't significantly slower in the grand scheme of things.